README.md
# The Learn Storybook design system

The Learn Storybook design system is a subset of the full [Storybook design system](https://github.com/storybookjs/design-system/), created as a learning resource for those interested in learning how to write and publish a design system using best in practice techniques.

Learn more at [Learn Storybook](https://learnstorybook.com).

hehe


CHANGELOG.md
# v0.1.34 (Tue Mar 31 2020)

#### 🐛 Bug Fix

- set primary color correctly again after demo [#37](https://github.com/johnlyden/learnstorybook-design-system/pull/37) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.33 (Tue Mar 31 2020)

#### 🐛 Bug Fix

- update primary color to purple [#36](https://github.com/johnlyden/learnstorybook-design-system/pull/36) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.32 (Tue Mar 31 2020)

#### 🐛 Bug Fix

- add back default button color" [#35](https://github.com/johnlyden/learnstorybook-design-system/pull/35) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.31 (Tue Mar 31 2020)

#### 🐛 Bug Fix

- update primary color [#34](https://github.com/johnlyden/learnstorybook-design-system/pull/34) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.30 (Tue Mar 31 2020)

#### 🐛 Bug Fix

- Export link [#33](https://github.com/johnlyden/learnstorybook-design-system/pull/33) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.29 (Tue Mar 31 2020)

#### 🐛 Bug Fix

- export link [#32](https://github.com/johnlyden/learnstorybook-design-system/pull/32) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.28 (Tue Mar 31 2020)

#### 🐛 Bug Fix

- Blah [#31](https://github.com/johnlyden/learnstorybook-design-system/pull/31) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.27 (Mon Mar 30 2020)

#### 🐛 Bug Fix

- Blah [#30](https://github.com/johnlyden/learnstorybook-design-system/pull/30) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.26 (Mon Mar 30 2020)

#### 🐛 Bug Fix

- add some other pages - organize [#29](https://github.com/johnlyden/learnstorybook-design-system/pull/29) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.25 (Sun Mar 29 2020)

#### 🐛 Bug Fix

- add other back up fonts [#28](https://github.com/johnlyden/learnstorybook-design-system/pull/28) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.24 (Sun Mar 29 2020)

#### 🐛 Bug Fix

- fix icon export [#27](https://github.com/johnlyden/learnstorybook-design-system/pull/27) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.23 (Sun Mar 29 2020)

#### 🐛 Bug Fix

- tryin this [#26](https://github.com/johnlyden/learnstorybook-design-system/pull/26) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.22 (Sun Mar 29 2020)

#### 🐛 Bug Fix

- adding font configs [#25](https://github.com/johnlyden/learnstorybook-design-system/pull/25) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.21 (Sun Mar 29 2020)

#### 🐛 Bug Fix

- add markdown story [#24](https://github.com/johnlyden/learnstorybook-design-system/pull/24) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.20 (Sat Mar 28 2020)

#### 🐛 Bug Fix

- again [#23](https://github.com/johnlyden/learnstorybook-design-system/pull/23) ([@johnlyden](https://github.com/johnlyden))
- plock [#22](https://github.com/johnlyden/learnstorybook-design-system/pull/22) ([@johnlyden](https://github.com/johnlyden))
- move buid [#21](https://github.com/johnlyden/learnstorybook-design-system/pull/21) ([@johnlyden](https://github.com/johnlyden))
- add cpx - cop fopy fonts into package [#20](https://github.com/johnlyden/learnstorybook-design-system/pull/20) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.19 (Fri Mar 27 2020)

#### 🐛 Bug Fix

- gonna build before publish [#19](https://github.com/johnlyden/learnstorybook-design-system/pull/19) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.18 (Fri Mar 27 2020)

#### 🐛 Bug Fix

- remove files [#18](https://github.com/johnlyden/learnstorybook-design-system/pull/18) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.17 (Fri Mar 27 2020)

#### 🐛 Bug Fix

- yo [#17](https://github.com/johnlyden/learnstorybook-design-system/pull/17) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.16 (Fri Mar 27 2020)

#### 🐛 Bug Fix

- remove build [#16](https://github.com/johnlyden/learnstorybook-design-system/pull/16) ([@johnlyden](https://github.com/johnlyden))
- Remove .npmignore [#15](https://github.com/johnlyden/learnstorybook-design-system/pull/15) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.15 (Fri Mar 27 2020)

#### 🐛 Bug Fix

- Remove badge [#14](https://github.com/johnlyden/learnstorybook-design-system/pull/14) ([@johnlyden](https://github.com/johnlyden))

#### Authors: 1

- John Lyden ([@johnlyden](https://github.com/johnlyden))

---

# v0.1.14 (Tue Mar 24 2020)

#### 🐛 Bug Fix


src/shared/styles.js
import { css } from 'styled-components';

// Global style variables
export const background = {
  app: '#F6F9FC',
  appInverse: '#7A8997',
  positive: '#E1FFD4',
  negative: '#FEDED2',
  warning: '#FFF5CF',
};

export const color = {
  // Palette
  // primary: '#4ba373', // green1
  primary: '#6F2CAC',
  secondary: '#008b84', // ocean
  tertiary: '#DDDDDD',

  orange: '#FC521F',
  gold: '#FFAE00',
  green: '#66BF3C',
  seafoam: '#37D5D3',
  purple: '#6F2CAC',
  ultraviolet: '#2A0481',

  // Monochrome
  lightest: '#FFFFFF',
  lighter: '#F8F8F8',
  light: '#F3F3F3',
  mediumlight: '#EEEEEE',
  medium: '#DDDDDD',
  mediumdark: '#999999',
  dark: '#666666',
  darker: '#444444',
  darkest: '#333333',

  border: 'rgba(0,0,0,.1)',

  // Status
  positive: '#66BF3C',
  negative: '#FF4400',
  warning: '#E69D00',

  // links
  william: '#406566',
  cobalt: '#006298',
  activeBlue: '#0083cc',
  focusBlue: '#00314d',
  manatee: '#919193',
};

export const spacing = {
  padding: {
    small: 10,
    medium: 20,
    large: 30,
  },
  borderRadius: {
    small: 5,
    default: 10,
  },
};

export const typography = {
  type: {
    primary: '"Nexa-Regular", "Helvetica Neue", Helvetica, Arial, sans-serif',
    bold: '"Nexa-Bold", "Helvetica Neue", Helvetica, Arial, sans-serif',
    light: '"Nexa-Light", "Helvetica Neue", Helvetica, Arial, sans-serif',
    regularItalic:
      '"Nexa-Regular-Italic", "Helvetica Neue", Helvetica, Arial, sans-serif',
    code:
      '"SFMono-Regular", Consolas, "Liberation Mono", Menlo, Courier, monospace',
  },
  weight: {
    regular: '400',
    bold: '700',
    extrabold: '800',
    black: '900',
  },
  size: {
    s1: '12',
    s2: '14',
    s3: '16',
    m1: '20',
    m2: '24',
    m3: '28',
    l1: '32',
    l2: '40',
    l3: '48',
    code: '90',
    _s1: '18.7',
  },
};

export const breakpoint = 600;
export const pageMargin = '5.55555';

export const pageMargins = css`
  padding: 0 ${spacing.padding.medium}px;
  @media (min-width: ${breakpoint * 1}px) {
    margin: 0 ${pageMargin * 1}%;
  }
  @media (min-width: ${breakpoint * 2}px) {
    margin: 0 ${pageMargin * 2}%;
  }
  @media (min-width: ${breakpoint * 3}px) {
    margin: 0 ${pageMargin * 3}%;
  }
  @media (min-width: ${breakpoint * 4}px) {
    margin: 0 ${pageMargin * 4}%;
  }
`;
